controllers/clusters/helpers.go

```diff
@@ -280,3 +283,57 @@ type Object interface {
 	client.Object
 	NewPatch() client.Patch
 }
+
+type ExternalChanger[T any] interface {
+	Object
+	GetSpec() T
+	IsSpecEqual(T) bool
+	GetClusterID() string
+}
+
+func handleExternalChanges[T any](
+	r record.EventRecorder,
+	c client.Client,
+	resource, iResource ExternalChanger[T],
+	l logr.Logger,
+) (reconcile.Result, error) {
+	patch := resource.NewPatch()
+
+	if !resource.IsSpecEqual(iResource.GetSpec()) {
+		resource.GetAnnotations()[models.ExternalChangesAnnotation] = models.True
+
+		l.Info(msgExternalChanges,
+			"specification of k8s resource", resource.GetSpec(),
+			"data from Instaclustr ", iResource.GetSpec())
+
+		msgDiffSpecs, err := createSpecDifferenceMessage(resource.GetSpec(), iResource.GetSpec())
+		if err != nil {
+			l.Error(err, "Cannot create specification difference message",
+				"instaclustr data", iResource.GetSpec(), "k8s resource spec", resource.GetSpec())
+			return models.ExitReconcile, nil
+		}
+
+		r.Eventf(resource, models.Warning, models.ExternalChanges, msgDiffSpecs)
+	} else {
+		resource.GetAnnotations()[models.ExternalChangesAnnotation] = ""
+		resource.GetAnnotations()[models.ResourceStateAnnotation] = models.UpdatedEvent
+
+		fmt.Println("resource.GetAnnotations()[models.ResourceStateAnnotation] ", resource.GetAnnotations()[models.ResourceStateAnnotation])
+
+		l.Info("External changes have been reconciled", "resource ID", resource.GetClusterID())
```

Comment:
I think `resource ID` -> `cluster ID` is more suitable

Reply:
fixed